### PR TITLE
Add release checklist

### PR DESCRIPTION
**Pull request by andrewfrench**
## The Problem/Issue/Bug:
The steps involved in generating a `ddev` release weren't version controlled and relied on institutional knowledge.

## How this PR Solves The Problem:
Add an authoritative workflow to create a `ddev` release.

### CODE REVIEW

**Comment by andrewfrench** [comment]
Sounds good -- this touches no code so it's a great sanity check for the buildkite PR. I'll hold off on pulling until we get a green status.

File: docs/developers/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+## `ddev` Release Checklist 
+
+- [ ] [Create a release](https://github.com/drud/ddev/releases/new) using the github web UI. This will initiate a CircleCI tag build.
+- [ ] Check out the ddev source with `git fetch upstream && git checkout <release_tag>`
+- [ ] In each directory if containers/ do a `make push`.
```
Comment (rickmanelius) [inline docs/developers/release-checklist.md]: I'm assuming "containers/" would be a directory?

File: docs/developers/release-checklist.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- [ ] Add the commit list to the release page
+- [ ] Download artifacts from CircleCI and upload them (except for the Windows installer and its sha256.txt) to the release page
+- [ ] Update the `ddev` [Homebrew formula](https://github.com/drud/homebrew-ddev) with the MacOS `.tar.gz` and SHA checksum
+- [ ] Test `brew upgrade ddev` and make sure ddev is the right version and behaves well.
+- [ ] Sign the Windows installer and create a new sha256.txt according to steps below.
```
Comment (rickmanelius) [inline docs/developers/release-checklist.md]: Small nit, but getting consistent periods at the end of each sentence would be great!

**Comment by andrewfrench** [comment]
Agreed @rickmanelius, just waiting on #902 then I'll rebase for testing.

**Comment by rfay** [comment]
I took the liberty of rebasing this @andrewfrench , trying to get all the PRs I've blocked up out of the way (and see how our testing goes).

**Comment by andrewfrench** [comment]
Woo! Fingers crossed.

**Pull request merged**